__init__.py

```diff
@@ -0,0 +1,430 @@
+    for w in words:
+        if not w:
+            continue
+        res = re.search(r"\b" + re.escape(w) + r"\b", sentence)
+        if res:
+            return res
+    return False
+
+
+# TODO: Move the mycroft.util.parse
+def fuzzy_match(x, against):
+    from difflib import SequenceMatcher
+    # Returns a value 0.0 - 1.0
+    return SequenceMatcher(None, x, against).ratio()
+
+
+# TODO: Move to mycroft.util.time
+def to_timestamp(dt_utc):
+    # return epoch (milliseconds since 1970-1-1) as float
+    return (dt_utc - datetime.datetime(1970, 1, 1)).total_seconds()
+
+
+# TODO: Move to mycroft.util.time
+def from_timestamp(epoch_utc):
+    # return a UTC datatime from an epoch millisecond count
+    epoch_origin = datetime.datetime(1970, 1, 1)
+    return epoch_origin + datetime.timedelta(0, epoch_utc)
+
+# For now, using the Wink V1 API which won't require anything except the
+# username and password.  For details, see:
+# https://wink.docs.apiary.io/#reference
+
+# These normally shouldn't be published, but Quirky released these
+# accidentally years ago and have left them alive.  Use these until
+# we switch to the Quirky v2 API
+#
+client_id = 'quirky_wink_ios_app'
+client_secret = 'ce609edf5e85015d393e7859a38056fe'
+
+
+class WinkIoTSkill(MycroftSkill):
+    debug_level = 3  # 0-10, 10 showing the most debugging info
+
+    def __init__(self):
+        super(WinkIoTSkill, self).__init__(name="WinkIoTSkill")
+        self.settings["access_token"] = None
+        self.settings["refresh_token"] = None
+        self.settings["token_expiration"] = to_timestamp(
+                                                datetime.datetime.utcnow())
+        self.settings['email'] = ""
+        self.settings['password'] = ""
+        self._device_cache = None
+        self._group_cache = None
+        self._active_room = None
+
+    def debug(self, message, level=1, char=None):
+        # Debugging assistance.
+        # Lower number are More important.  Only number less than the
+        # debug_level get logged.  Indentation is the inverse of this,
+        # with level 0 having 10 dashes as a prefix, 1 have 9, etc.
+        if level > WinkIoTSkill.debug_level:
+            return
+
+        if not char:
+            char = "-"
+        self.log.debug(char*(10-level) + " " + message)
+
+    # TODO: Move in to MycroftSkill
+    def translate_namedvalues(self, name, delim=None):
+        """
+        Load translation dict containing names and values.
+
+        This loads a simple CSV from the 'dialog' folders.
+        The name is the first list item, the value is the
+        second.  Lines prefixed with # or // get ignored
+
+        Args:
+            name (str): name of the .value file, no extension needed
+            delim (char): delimiter character used, default is ','
+
+        Returns:
+            dict: name and value dictionary, or [] if load fails
+        """
+        import csv
+        from os.path import join
```

Comment:
i think this would throw pep 8 errors but is this done here so you can remember to move them to a different module in mycroft-core?

Reply:
Yep, I intend to make that part of mycroft-core